src/copy.jl

```diff
@@ -90,3 +90,222 @@ function defaultcopy!(dest::MOI.AbstractInstance, src::MOI.AbstractInstance)
 
     return MOI.CopyResult(MOI.CopySuccess, "", idxmap)
 end
+
+"""
+    allocatevariables!(instance::MOI.AbstractInstance, nvars::Integer)
+
+Creates `nvars` variables and returns a vector of `nvars` variable indices.
+"""
+function allocatevariables! end
+
+"""
+    allocate!(instance::AbstractInstance, attr::AbstractInstanceAttribute, value)
+    allocate!(instance::AbstractInstance, attr::AbstractVariableAttribute, v::VariableIndex, value)
+    allocate!(instance::AbstractInstance, attr::AbstractConstraintAttribute, c::ConstraintIndex, value)
+
+Informs `instance` that `load!` will be called with the same arguments after `loadvariables!` is called.
```

Comment:
`allocate!(instance::AbstractInstance, attr::AbstractConstraintAttribute, c::ConstraintIndex, value)` isn't called right after `loadvariables!`.

Reply:
No, I don't think the implementation should assume anything about the order of the calls after loadvariables. the fact that loadvariables is the first that is called among the `load...` calls should be enough.

Reply:
It should be clearer now with the comment added in 8b3db80